### DISCUSSION TITLE
I'm not getting a signal

### DISCUSSION THREAD

**Discussion by khromenokroman** [Kistler-Group/sdbus-cpp#449] [Q&A]
used lib 1.4.0

busctl
```
‣ Type=signal  Endian=l  Flags=1  Version=1 Cookie=2950  Timestamp="Fri 2024-07-12 05:12:32.158034 UTC"
  Sender=:1.67774  Path=/ru/test/management/converter  Interface=ru.test.management.converter.modules  Member=config_changed
  UniqueName=:1.67774
  MESSAGE "" {
  };
```

run
```
#include <iostream>
#include <sdbus-c++/sdbus-c++.h>

int main() {
    std::cout << "Start prog" << std::endl;

    auto converter_signal_proxy = sdbus::createProxy("ru.test.management",
                                                     "/ru/test/management/converter");
    converter_signal_proxy->uponSignal("config_changed").onInterface("ru.test.management.converter.modules")
            .call([]() {
                std::cout << "Catch signal converter" << std::endl;
            });

    sleep(15);

    std::cout << "End prog" << std::endl;

    return 0;
}
```
output
```
Start prog
End prog
```

the monitor shows that there is a signal, but I can't get it with my application, what am I doing wrong

I did it according to the documentation https://github.com/Kistler-Group/sdbus-cpp/blob/master/docs/using-sdbus-c++.md

Client side

neeed help :))

but if you do it like this, I get a signal

```
#include <dbus/dbus.h>
#include <cstdio>

[[noreturn]] void listen_to_signals(DBusConnection* conn) {
    dbus_bus_add_match(conn, "type='signal',interface='ru.test.management.converter.modules'", nullptr);
    dbus_connection_flush(conn);

    while(true) {
        DBusMessage* msg;
        dbus_connection_read_write(conn, 0);
        msg = dbus_connection_pop_message(conn);

        if (msg != nullptr) {
            if (dbus_message_is_signal(msg, "ru.test.management.converter.modules", "config_changed")) {
                printf("Received Signal\n");
            }
            dbus_message_unref(msg);
        }
    }
}

int main() {
    DBusError err;
    DBusConnection* conn;

    dbus_error_init(&err);

    conn = dbus_bus_get(DBUS_BUS_SYSTEM, &err);
    if (dbus_error_is_set(&err)) {
        fprintf(stderr, "Connection Error (%s)\n", err.message);
        dbus_error_free(&err);
        return 1;
    }

    listen_to_signals(conn);
}
```

**Comment by khromenokroman** [comment]
@sangelovic neeed heelp :)

**Comment by sangelovic** [comment]
@khromenokroman You did not really do that according to the documentation ;-) Your example is missing `converter_signal_proxy->finishRegistration();` call after the signal calback handler is installed.

This is required in sdbus-c++ v1.x releases (you're saying you're using v1.4).

This has changed since sdbus-c++ v2.0, the registration does not require `finishRegistration()` in sdbus-c++ v2.

Did this help?

**Reply by khromenokroman** [reply]
> @khromenokroman You did not really do that according to the documentation ;-) Your example is missing `converter_signal_proxy->finishRegistration();` call after the signal calback handler is installed.
> 
> This is required in sdbus-c++ v1.x releases (you're saying you're using v1.4).
> 
> This has changed since sdbus-c++ v2.0, the registration does not require `finishRegistration()` in sdbus-c++ v2.
> 
> Did this help?

@sangelovic yes, you are absolutely right, I did not pay attention to the fact that the old version may be different, I apologize

` Did this help?` yes, it's working, thank you very much :)

**Reply by sangelovic** [reply]
Glad to hear. Just use sdbus-c++ v2.0 if you can, it has a number of API improvements, including the above mentioned one.

**Reply by khromenokroman** [reply]
> Glad to hear. Just use sdbus-c++ v2.0 if you can, it has a number of API improvements, including the above mentioned one.

@sangelovic Thank you, I will try to do it
